libyul/AsmAnalysis.h

```diff
@@ -112,6 +113,7 @@ class AsmAnalyzer
 	/// Variables that are active at the current point in assembly (as opposed to
 	/// "part of the scope but not yet declared")
 	std::set<Scope::Variable const*> m_activeVariables;
+	std::vector<YulString> m_typeOfCurrentExpression;
```

Comment:
I don't understand fully how this vector is being used atm. Can you explain a bit?